conanfile.py

```diff
@@ -148,6 +148,7 @@ def generate(self):
         tc.variables["MP_UNITS_BUILD_LA"] = self._build_all and not self._skip_la
         tc.variables["MP_UNITS_USE_LIBFMT"] = self._use_libfmt
         if self._build_modules:
+            tc.variables["CMAKE_CXX_COMPILER_CLANG_SCAN_DEPS"] = "/usr/bin/clang-scan-deps-16"
```

Comment:
This looks fishy...

Is there any way to expose that through the CMake environment variables? Hardcoding this in a conanfile that might also be used on Windows seems wrong.

Reply:
I agree.

First of all, this doesn't just work because the installed Clang package doesn't provide `clang-scan-deps` in `PATH`.
[`find_program`][] was probably used internally to do that,
and whatever its heuristics, didn't work for this.

[`find_program`]: https://cmake.org/cmake/help/latest/command/find_program.html

Normally, you'd just define this variable in a preset for CI or the CMake's command line.
You have an indirection through Conan, so I have no idea what's the equivalent way.

Reply:
I think that a much better solution is to provide a custom CMake toolchain file with those variables set for CI and then call Conan in CI with the following:

```
conan -c tools.cmake.cmaketoolchain:user_toolchain=<path_to_toolchain> ...
```

This will prepend the Conan-generated toolchain file with the one provided by the user.

Reply:
Good.
Now we just need to find out the path of the scanner
as installed by the Clang package.

Reply:
```
$ which clang-scan-deps-16
/usr/bin/clang-scan-deps-16
```

Maybe we should generate this file on the fly in CI, as different versions of clang will require different variable values.

Reply:
I already tried that (<https://github.com/mpusz/mp-units/actions/runs/6304075675/job/17114651127#step:14:692>):
> ` /bin/sh: 1: /usr/bin/clang-scan-deps-16: not found`

Reply:
This should help: https://github.com/mpusz/mp-units/commit/993ea8e3ac8622f1f6647ae51a27cc256b60057e. It can be conditionally installed only when clang modules are needed.

Reply:
You can easily add Conan `-c` options via "conan-config" GitHub Actions matrix parameter in _ci-conan.yml_.